samples/android/app/src/androidTest/java/org/mozilla/samples/gleancore/pings/BackgroundPingTest.kt

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.samples.gleancore.pings
+
+import android.content.Context
+import android.content.Intent
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.ServiceTestRule
+import mozilla.telemetry.glean.testing.GleanTestLocalServer
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertTrue
+import org.junit.Ignore
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.samples.gleancore.SampleBackgroundProcess
+
+// Note that this is ignored because for some reason the mock server
+// doesn't seem to be able to catch the background ping when it is
+// sent, despite the logs clearly stating that the ping is being sent
+// and from testing with the Glean Debug View also clearly showing
+// that the ping is being sent.
```

Comment:
`GleanTestLocalServer` calls `Glean.testSetLocalEndpoint(localPort)`, which of course only happens in the main process then.
You probably would need to change the config manually for the subprocess: https://github.com/mozilla/glean/blob/main/glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt#L588-L591

Reply:
Ha, worth a shot

Reply:
Bug 1846518 filed to follow up on this, as it turned out to be non-trivial and we are due a Glean release for many things